app/services/open_humans_service.rb

```diff
@@ -2,110 +2,120 @@
 
 class OpenHumansService
   # used to do the interfacing w/ Open Humans
+  BASE_API_URL = 'https://www.openhumans.org/api/direct-sharing/project'
+  BOUNDARY = '0P3NSNPH34RT50PENHUM4N5'
 
-  def get_access_tokens(user, code)
+  def initialize(user)
+    @user = user
+    @oh_profile = @user.open_humans_profile
+  end
+
+  def authenticate(code)
+    # lets convert
+    get_access_tokens(code)
+    set_open_humans_ids
+    # delete old files if there are any
+    begin
+      delete_opensnp_id
+    end
+    upload_opensnp_id
+  end
+
+  def get_access_tokens(code)
     # authenticate w/ our client id/secret against API
     # post with the key the user provided us with.
     url = URI.parse('https://www.openhumans.org/oauth2/token/')
     req = Net::HTTP::Post.new(url.request_uri)
-    req.basic_auth ENV.fetch('OH_client_id'), ENV.fetch('OH_client_secret')
+    req.basic_auth ENV.fetch('OH_CLIENT_ID'), ENV.fetch('OH_CLIENT_SECRET')
     req.set_form_data('grant_type' => 'authorization_code',
                       'code' => code,
-                      'redirect_uri' => 'http://localhost:3000/openhumans/authorize')
-    # set up request to use https
+                      'redirect_uri' => OH_REDIRECT_URL)
```

Comment:
You can probably use `openhumans_authorize_path` here, right?

Reply:
Mh… I don't exactly get what you mean

Reply:
`OH_REDIRECT_URL` contains the full URL to `/openhumans/authorize`, which is what the URL helpers give you, which are generated from the routes. `openhumans_authorize_path` gives you the relative URL, `openhumans_authorize_url` the absolute one. For the latter to work, you need to configure the domain via the `default_url_options` in the environments:

```ruby
config.action_controller.default_url_options = { host: 'myproduction.com' }
```